File: OrchidCore/src/main/java/com/eden/orchid/api/options/globals/SiteGlobal.java
package com.eden.orchid.api.options.globals;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.options.TemplateGlobal;
import com.eden.orchid.api.site.OrchidSite;

public class SiteGlobal implements TemplateGlobal<OrchidSite> {

    @Override
    public String key() {
        return "site";
    }

    @Override
    public OrchidSite get(OrchidContext context) {
        return context.getSite();
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/api/options/globals/SiteGlobal.kt
package com.eden.orchid.api.options.globals

import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.TemplateGlobal
import com.eden.orchid.api.site.OrchidSite
import com.eden.orchid.api.theme.pages.OrchidPage

class SiteGlobal : TemplateGlobal<OrchidSite?> {
    override fun key() = "site"
    override fun get(context: OrchidContext, page: OrchidPage?): OrchidSite? = context.site
}
